core/sail/fts/lucene/src/test/java/org/eclipse/rdf4j/sail/lucene/AbstractGenericLuceneTest.java

```diff
@@ -0,0 +1,914 @@
+			TupleQuery query = connection.prepareTupleQuery(QueryLanguage.SERQL, serql);
+			TupleQueryResult tqr = query.evaluate();
+
+			// check the results
+			int actualResults = 0;
+			Set<Integer> matched = new HashSet<Integer>();
+			while (tqr.hasNext()) {
+				BindingSet bs = tqr.next();
+				actualResults++;
+
+				boolean matches;
+				for (int resultSetID = 0; resultSetID < expectedResultSet.size(); resultSetID++) {
+					// ignore results that matched before
+					if (matched.contains(resultSetID))
+						continue;
+
+					// assume it matches
+					matches = true;
+
+					// get the result we compare with now
+					Map<String, String> expectedResult = new HashMap<String, String>(
+							expectedResultSet.get(resultSetID));
+
+					// get all var names
+					Collection<String> vars = new ArrayList<String>(expectedResult.keySet());
+
+					// check if all actual results are expected
+					for (String var : vars) {
+						String expectedVal = expectedResult.get(var);
+						Value actualVal = bs.getValue(var);
+
+						if (expectedVal == null) {
+							// don't care about the actual value, as long as there is
+							// one
+							if (actualVal == null) {
+								matches = false;
+								break;
+							}
+						}
+						else {
+							// compare the values
+							if ((actualVal == null)
+									|| (expectedVal.compareTo(actualVal.stringValue()) != 0))
+							{
+								matches = false;
+								break;
+							}
+						}
+
+						// remove the matched result so that we do not match it twice
+						expectedResult.remove(var);
+					}
+
+					// check if expected results were existing
+					if (expectedResult.size() != 0) {
+						matches = false;
+					}
+
+					if (matches) {
+						matched.add(resultSetID);
+						break;
+					}
+				}
+			}
+			tqr.close();
+
+			// the number of matched expected results must be equal to the number
+			// of actual results
+			assertEquals("How many expected results were retrieved for query #" + queryID + "?",
+					expectedResultSet.size(), matched.size());
+			assertEquals("How many actual results were retrieved for query #" + queryID + "?",
+					expectedResultSet.size(), actualResults);
+		}
+	}
+
+	@Test
+	public void testPredicateLuceneQueries()
+		throws MalformedQueryException, RepositoryException, QueryEvaluationException
+	{
+		// prepare the query
+		String[] queries = new String[] {
+				"SELECT \n" + "  Resource, Score, Snippet \n" + "FROM \n" + "  {Resource} <" + MATCHES
+						+ "> {} \n" + "    <" + QUERY + "> {\"one\"}; \n" + "    <" + SCORE + "> {Score}; \n"
+						+ "    <" + SNIPPET + "> {Snippet}",
+				"SELECT \n" + "  Resource, Score, Snippet \n" + "FROM \n" + "  {Resource} <" + MATCHES
+						+ "> {} \n" + "    <" + QUERY + "> {\"five\"}; \n" + "    <" + SCORE + "> {Score}; \n"
+						+ "    <" + SNIPPET + "> {Snippet}" };
+
+		ArrayList<List<Map<String, String>>> results = new ArrayList<List<Map<String, String>>>();
+		ArrayList<Map<String, String>> resultSet = null;
```

Comment:
Do not reuse the same variable names for different assertions, it makes it difficult to verify whether there are side effects between assertions.